pkg/pgmodel/querier/timestamp_series.go

```diff
@@ -0,0 +1,61 @@
+package querier
+
+import (
+	"time"
+
+	"github.com/jackc/pgtype"
+	prommodel "github.com/prometheus/common/model"
+	"github.com/timescale/promscale/pkg/pgmodel/model"
+)
+
+// TimestampSeries represent an array of timestamps (model.Time/int64) that is 0-indexed.
+type TimestampSeries interface {
+	//At returns the element at an index location, as well as a bool to indicate
+	//whether the value is valid (or NULL for example)
+	At(index int) (int64, bool)
+	Len() int
+}
+
+//rowTimestampSeries is a TimestampSeries based on data fetched from a database row
+type rowTimestampSeries struct {
+	times *pgtype.TimestamptzArray
+}
+
+func NewRowTimestampSeries(times *pgtype.TimestamptzArray) *rowTimestampSeries {
```

Comment:
Do we need this exported? If I am not wrong, its being used in the querier package only. Same applies to `NewRegularTimestampSeries()` too.